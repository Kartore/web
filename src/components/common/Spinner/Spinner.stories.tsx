import { Spinner } from '.';
import { Meta, StoryObj } from '@storybook/react';

const meta = {
  component: Spinner,
} satisfies Meta<typeof Spinner>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
