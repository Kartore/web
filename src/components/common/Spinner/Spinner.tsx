import { ComponentPropsWithoutRef, forwardRef } from 'react';
import { tv, VariantProps } from 'tailwind-variants';

const spinnerVariant = tv({
  base: 'inline-block aspect-square h-[1em] w-[1em] animate-spin rounded-full border-[0.2em] border-current border-r-transparent  border-t-transparent animation-duration-[600ms]',
  variants: {},
});

type SpinnerVariantProps = VariantProps<typeof spinnerVariant>;

type SpinnerProps = ComponentPropsWithoutRef<'div'> & SpinnerVariantProps;

export const Spinner = forwardRef<HTMLDivElement, SpinnerProps>((props, ref) => {
  return <div {...props} className={spinnerVariant(props)} ref={ref} />;
});

Spinner.displayName = 'Spinner';
