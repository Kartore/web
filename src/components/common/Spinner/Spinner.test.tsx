import { describe } from 'vitest';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

import * as Stories from './Spinner.stories';

const { Default } = composeStories(Stories);

describe('Component: Spinner', () => {
  describe('Snapshot', () => {
    it('Default', () => {
      const { asFragment } = render(<Default />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
