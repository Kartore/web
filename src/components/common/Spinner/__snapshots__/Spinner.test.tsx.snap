// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: Spinner > Snapshot > Default 1`] = `
<DocumentFragment>
  <div
    class="inline-block aspect-square h-[1em] w-[1em] animate-spin rounded-full border-[0.2em] border-current border-r-transparent border-t-transparent animation-duration-[600ms]"
  />
</DocumentFragment>
`;
